### PR TITLE
add support for cfsec

### PR DESCRIPTION
**Pull request by owenrumney**
- CFSec integration to scan config files that are typed as Cloudformation
- add test cases as per terraform ones

### CODE REVIEW

**Review by liamg — APPROVED** [review]

File: config/scanner/scanner.go [hunk 1]
```diff
@@ -69,6 +78,13 @@ func (s Scanner) ScanConfigs(ctx context.Context, files []types.Config) ([]types
 	}
 	misconfs = append(misconfs, results...)
 
+	// Scan terraform files by CFSec
```
Comment (knqyf263) [inline config/scanner/scanner.go:81]:
```suggestion
	// Scan CloudFormation files by CFSec
```

**Review by knqyf263 — COMMENTED** [review]
It looks good! Did you confirm this commit work with Trivy? The following commands should work.

```
$ trivy conf /path/to/cf_dir
```

```
$ trivy fs ----security-checks config /path/to/cf_dir
```

Also, I'd like you to test the directory mixing Kubernetes, Terraform, and CloudFormation files.

**Comment by owenrumney** [comment]
> It looks good! Did you confirm this commit work with Trivy? The following commands should work.
> 
> ```
> $ trivy conf /path/to/cf_dir
> ```
> 
> ```
> $ trivy fs ----security-checks config /path/to/cf_dir
> ```
> 
> Also, I'd like you to test the directory mixing Kubernetes, Terraform, and CloudFormation files.

I realise I assumed the analysis of files was done in trivy and I would do that as the next step. I see that this is done in Fanal now so I need to refactor so that CloudFormation doesn't get mistaken for normal Yaml or Json. Will mark as draft while investigating.

**Comment by owenrumney** [comment]
cfsec working from Trivy

image

**Comment by owenrumney** [comment]
Hey @knqyf263  - I've tested this in Trivy using both `conf /path` and `--security-checks config` and they both work.

If you're happy to approve this I'll do a PR to update Trivy?

**Review by liamg — APPROVED** [review]

**Comment by knqyf263** [comment]
>If you're happy to approve this I'll do a PR to update Trivy?

Yes, please!

**Comment by owenrumney** [comment]
> > If you're happy to approve this I'll do a PR to update Trivy?
> 
> Yes, please!

If you can approve I'll grab the commit and update

**Pull request merged**